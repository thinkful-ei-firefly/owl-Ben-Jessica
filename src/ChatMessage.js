import React from 'react';
import './chatItem.css';

function ChatMessage(props) {
  
  const time = new Date(props.time);
  const hours = time.getHours();
  const minutes = time.getMinutes();
  return (
    <div id={props.id} className="chatItem">
      <img src={props.avatar} className="chatIcon" alt="icon" />
      <span className="personName">{props.name} </span>
      <span className="time">{hours}:{minutes}</span>
      <p>{props.message}</p>
    </div>
  );
}

export default ChatMessage;
