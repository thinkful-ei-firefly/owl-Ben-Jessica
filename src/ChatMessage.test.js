import React from 'react';
import ReactDOM from 'react-dom';
import store from './store'
import renderer from 'react-test-renderer';

import ChatMessage from './ChatMessage';

describe('<Stage />', () => {
    const part = store.participants;
    const list = store.chatEvents;
    const person = list[0];
    
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<ChatMessage
            key={person.participantId}
            avatar={part.find(item => item.id === person.participantId).avatar}
            name={part.find(item => item.id === person.participantId).name}
            message={person.message}
            time={person.timestamp}
          />, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        // Render the component, as JSON
        const tree = renderer.create(<ChatMessage
            key={person.participantId}
            avatar={part.find(item => item.id === person.participantId).avatar}
            name={part.find(item => item.id === person.participantId).name}
            message={person.message}
            time={person.timestamp}
          />).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});