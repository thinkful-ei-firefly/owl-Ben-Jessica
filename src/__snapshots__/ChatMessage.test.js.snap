// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage /> renders this UI as expected 1`] = `
<div
  className="chatItem"
>
  <img
    alt="icon"
    className="chatIcon"
    src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
  />
  <span
    className="personName"
  >
    Axe Kubicka
     
  </span>
  <span
    className="time"
  >
    7
    :
    48
  </span>
  <p>
    Hello world
  </p>
</div>
`;
